src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseRepositoryTests.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Mappings;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using Xunit;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Tests.Services.DbUtils;
+using static GovUk.Education.ExploreEducationStatistics.Data.Model.Database.StatisticsDbUtils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Tests.Services
+{
+    public class ReleaseRepositoryTests
+    {
+        [Fact]
+        public async Task ReleaseHierarchyCreated()
+        {
+            var subjectFilename = "test-data.csv";
+            var subjectName = "Subject name";
+            var release = new Release
+            {
+                ReleaseName = "2000",
+                Publication = new Publication
+                {
+                    Title = "Test publication",
+                    Topic = new Topic
+                    {
+                        Id = Guid.NewGuid(),
+                        Title = "Test topic",
+                        Theme = new Theme
+                        {
+                            Id = Guid.NewGuid(),
+                            Title = "Test theme"
+                        }
+                    }
+                }
+            };
+            
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddAsync(release);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var statisticsDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbContextId))
+            {
+                var releaseRepository = new ReleaseRepository(contentDbContext, statisticsDbContext,
+                    Common.Services.MapperUtils.MapperForProfile<MappingProfiles>());
+
+                await releaseRepository.CreateReleaseAndSubjectHierarchy(
+                    release.Id,
+                    subjectFilename,
+                    subjectName);
+            }
+
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbContextId))
+            {
+                var themes = statisticsDbContext.Theme.ToList();
+                Assert.Single(themes);
+                Assert.Equal(release.Publication.Topic.Theme.Title, themes[0].Title);
```

Comment:
I think it would be good to confirm the id's here in stats are the same as they are in the content which could be done using `find` to get them by their content id and then asserting not null on that. This checks seems important to make sure that different id's are not being assigned. Also the set up of id's through this test class is a bit inconsistent, release and publication don't have them, but publication and topic do. If you don't assign any id's you will see the save of the content Release assigns the id's throughout the whole hierarchy.